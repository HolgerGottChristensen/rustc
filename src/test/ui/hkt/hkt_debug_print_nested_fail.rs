// HKT test that something cloneable can be called with clone, method call syntax, definition and call.
// check-fail

use std::fmt::{Debug};

struct Test;

fn test<I<%J: Debug>: Debug>(input: I<Test>) { //~ ERROR: `Test` doesn't implement `Debug`
    println!("Hej verden: {:?}", input);
}

fn main() {
    test::<Option<%J>>(Some(Test));
}
