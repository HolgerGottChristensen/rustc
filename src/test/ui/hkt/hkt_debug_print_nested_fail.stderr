error[E0277]: `Test` doesn't implement `Debug`
  --> $DIR/hkt_debug_print_nested_fail.rs:8:37
   |
LL | fn test<I<%J: Debug>: Debug>(input: I<Test>) {
   |                                     ^^^^^^^ `Test` cannot be formatted using `{:?}`
   |
   = help: the trait `Debug` is not implemented for `Test`
   = note: add `#[derive(Debug)]` to `Test` or manually `impl Debug for Test`
note: required by a bound in `test::I`
  --> $DIR/hkt_debug_print_nested_fail.rs:8:15
   |
LL | fn test<I<%J: Debug>: Debug>(input: I<Test>) {
   |               ^^^^^ required by this bound in `test::I`
help: consider annotating `Test` with `#[derive(Debug)]`
   |
LL | #[derive(Debug)]
   |

error: aborting due to previous error

For more information about this error, try `rustc --explain E0277`.
